packages/config-loader/src/lib/transform/types.ts

```diff
@@ -23,13 +23,8 @@ export type ReadFileFunc = (path: string) => Promise<string>;
 export type TransformFunc = (
   value: JsonValue,
   baseDir: string,
-) => Promise<
-  | {
-      applied: false;
-    }
-  | {
-      applied: true;
-      value: JsonValue | undefined;
-      newBaseDir?: string | undefined;
-    }
->;
+) => Promise<{
+  applied: boolean;
+  value?: JsonValue;
+  newBaseDir?: string;
+}>;
```

Comment:
Didn't spot this first time around, why is this change needed?

Reply:
It is simplifying the existing declaration into something that is functionally equivalent (no changes were made to any of the implementations of this type, and they all still compile).

Without this, there is some Typescript gymnastics that must be done for the `substitute` function in the `include.test.ts`.

With the previous declaration, `newBaseDir?: string | undefined;` compiled to `(string | undefined) | undefined`, and there were duplicate definitions of `applied: true` and `applied: false` (`applied: boolean`) that both had `value` and `newBaseDir` as `undefined` (either implicitly by leaving them out, or explicitly by declaring them as `string | undefined` or `JsonValue | undefined`).

If you would rather leave this type the same, I can revert and perform the necessary Typescript changes in `include.test.ts`, but I thought a refactor that simplified and didn't impact the functionality would be ok.  😄

Reply:
Would rather revert and do a bit of gymnastics in the test tbh. The old declaration guarantees makes it clearer that the extra fields aren't set if applied is `false`, and TypeScript might even be able to enforce that in some cases 😅

Reply:
Sure, no problem. Reverted.